cli/azd/pkg/apphost/generate.go

```diff
@@ -812,6 +935,37 @@ func (b *infraGenerator) Compile() error {
 		b.containerAppTemplateContexts[resourceName] = projectTemplateCtx
 	}
 
+	for moduleName, module := range b.bicepContext.BicepModules {
+		for paramName, paramValue := range module.Params {
+			// paramValue comes from the manifest and it can be any type, but it is saved into bicepContext with json.Marshal
```

Comment:
Let's extract this logic into its own method and then write some tests for it as well.